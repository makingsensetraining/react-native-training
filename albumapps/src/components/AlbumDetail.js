import React from 'react';
import { View, Text, Image, Linking } from 'react-native';
import { Card, CardSection, Button } from './common';

const styles = {
    headerContent: {
        flexDirection: 'column',
        justifyContent: 'space-around'
    },

    headerText: {
        fontSize: 18
    },

    thumbnail: {
        height: 50,
        width: 50,
    },

    thumbnailContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        marginLeft: 10,
        marginRight: 10
    },

    imageAlbum: {
        height: 300,
        flex: 1,
        width: null
    }
}

const AlbumDetail = ({ album }) => {

    const { title, artist, thumbnail_image, image, url } = album;
    const { thumbnail, headerContent, thumbnailContainer, headerText, imageAlbum } = styles;

    return(
        <Card>
            <CardSection>
                <View 
                    style={ thumbnailContainer }>
                    <Image 
                        style={thumbnail} 
                        source={{ uri: thumbnail_image }}
                        />
                </View>
                <View 
                    style={ headerContent }>
                    <Text style={ headerText }>{title}</Text>
                    <Text>{ artist }</Text>
                </View>
            </CardSection>
            
            <CardSection>
                <Image style={ imageAlbum } source={{ uri: image }}/>    
            </CardSection>

            <CardSection>
                <Button onPress={() => Linking.openURL(url)}>
                    Buy Now
                </Button>
            </CardSection>
        </Card>
    );
};

export default AlbumDetail;