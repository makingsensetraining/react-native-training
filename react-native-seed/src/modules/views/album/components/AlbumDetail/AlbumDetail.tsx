import React from 'react';
import Card from '../../../common/Card';
import { View, Text, Image } from 'react-native';
import CardSection from '../../../common/CardSection';

import { AlbumModel } from '../../../../models';

export interface IAlbumDetailProps {
  album: AlbumModel.IAlbum;
}

export default class AlbumDetail extends React.PureComponent<IAlbumDetailProps, {}> {

  public render() {
    const { album } = this.props;
    return (
      <Card>
        <CardSection>
          <View>
            <Image
              source={{ uri: album.thumbnail_image }}
            />
          </View>
          <View>
            <Text>{album.title}</Text>
            <Text>{album.artist}</Text>
          </View>
        </CardSection>

        <CardSection>
          <Image source={{ uri: album.image }} />
        </CardSection>
      </Card>
    );
  }
}
