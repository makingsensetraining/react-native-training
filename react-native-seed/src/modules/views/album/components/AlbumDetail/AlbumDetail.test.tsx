import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { getAlbum_1 } from '../../../../../test/entities';
import AlbumDetail, { IAlbumDetailProps } from './AlbumDetail';
import { any } from 'prop-types';

describe('AlbumDetail', () => {
  let props: IAlbumDetailProps;

  beforeEach(() => {

    AlbumDetail.linker = {
      canOpenURL: jest.fn().mockReturnValue(Promise.resolve(true)),
      openURL: jest.fn()
    } as any;

    props = {
      album: getAlbum_1()
    };

  });

  it('renders without crashing', () => {
    const rendered = renderer.create(<AlbumDetail {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should open a link', async () => {
    const url = 'https://www.amazon.com/Imaginaerum-Nightwish/dp/B006CBSE86';
    AlbumDetail.linker = {
      canOpenURL: jest.fn().mockReturnValue(Promise.resolve(true)),
      openURL: jest.fn()
    } as any;
    const wrapper = shallow(<AlbumDetail {...props} />);
    wrapper.setProps({ url });
    await (wrapper.find('Button').first().prop('onPress') as any)();
    expect(AlbumDetail.linker.openURL).toBeCalledWith(url);
  });
});
