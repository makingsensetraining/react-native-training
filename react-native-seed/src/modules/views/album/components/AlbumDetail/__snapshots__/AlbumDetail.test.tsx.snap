// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDetail renders without crashing 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://images-na.ssl-images-amazon.com/images/I/61R4EBzQVwL._SS500.jpg",
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "space-around",
          "marginLeft": 5,
        }
      }
    >
      &gt;
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        My album title
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Ferri Test
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://images-na.ssl-images-amazon.com/images/I/61R4EBzQVwL._SS500.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 300,
          "width": null,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#48BBEC",
          "borderColor": "#48BBEC",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 60,
          "justifyContent": "center",
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 5,
            "paddingTop": 5,
          }
        }
      >
        Buy Now
      </Text>
    </View>
  </View>
</View>
`;
